src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp

```diff
@@ -14,320 +14,173 @@
+            setup_channel->bindQueue(exchange_name, queue_name_, routing_keys[0], binding_arguments)
+            .onSuccess([&]
+            {
+                bindings_created = true;
+            })
+            .onError([&](const char * message)
+            {
+                bindings_error = true;
+                LOG_ERROR(log, "Failed to bind queue. Reason: {}", message);
+            });
+        }
+        else
+        {
+            /// Means there is only one queue with one consumer - no even distribution needed - no hash-exchange.
+            for (const auto & routing_key : routing_keys)
+            {
+                /// Binding directly to exchange, specified by the client.
+                setup_channel->bindQueue(exchange_name, queue_name_, routing_key)
                 .onSuccess([&]
                 {
                     bindings_created = true;
                 })
                 .onError([&](const char * message)
                 {
                     bindings_error = true;
-                    LOG_ERROR(log, "Failed to bind queue to key. Reason: {}", message);
+                    LOG_ERROR(log, "Failed to bind queue. Reason: {}", message);
                 });
             }
-            else
-            {
-                /// Means there is only one queue with one consumer - no even distribution needed - no hash-exchange.
-                for (const auto & routing_key : routing_keys)
-                {
-                    /// Binding directly to exchange, specified by the client.
-                    consumer_channel->bindQueue(exchange_name, queue_name_, routing_key)
-                    .onSuccess([&]
-                    {
-                        bindings_created = true;
-                    })
-                    .onError([&](const char * message)
-                    {
-                        bindings_error = true;
-                        LOG_ERROR(log, "Failed to bind queue to key. Reason: {}", message);
-                    });
-                }
-            }
         }
-    })
-    .onError([&](const char * message)
+    };
+
+    auto error_callback([&](const char * message)
     {
-        default_bindings_error = true;
+        bindings_error = true;
         LOG_ERROR(log, "Failed to declare queue on the channel. Reason: {}", message);
     });
 
-    /* Run event loop (which updates local variables in a separate thread) until bindings are created or failed to be created.
-     * It is important at this moment to make sure that queue bindings are created before any publishing can happen because
-     * otherwise messages will be routed nowhere.
-     */
-    while ((!default_bindings_created && !default_bindings_error) || (exchange_type_set && !bindings_created && !bindings_error))
+    AMQP::Table queue_settings;
+    if (!deadletter_exchange.empty())
+    {
+        queue_settings["x-dead-letter-exchange"] = deadletter_exchange;
```

Comment:
Maybe some comments about `dead-letter`?)

Reply:
`deadletter_exchange` is taken from table settings and therefore there is some explanation for it in docs, which I wrote. I think it is enough, isn't it?